src/main/scala/scorex/core/consensus/ModifierSemanticValidity.scala

```diff
@@ -1,7 +1,30 @@
 package scorex.core.consensus
 
-sealed trait ModifierSemanticValidity
-case object Absent extends ModifierSemanticValidity
-case object Unknown extends ModifierSemanticValidity
-case object Valid extends ModifierSemanticValidity
-case object Invalid extends ModifierSemanticValidity
\ No newline at end of file
+sealed trait ModifierSemanticValidity {
+  val code: Byte
+}
+
+object ModifierSemanticValidity {
+  def restoreFromCode(code: Byte): ModifierSemanticValidity = code match {
```

Comment:
maybe `restoreFromCode` ---> `fromCode` ?  `restore` does not add much information and makes the method name much longer.